core/instantiate.ml

```diff
@@ -298,22 +298,32 @@ let datatype = instantiate_datatype
 
 module SEnv = Env.String
 
-let apply_type : Types.datatype -> Types.type_arg list -> Types.datatype =
-  fun pt tyargs ->
+let type_arguments_to_instantiation_maps :
```

Comment:
I suppose the "proper OCaml name" would be `instantiation_maps_of_type_arguments`.